src/test/java/com/jfrog/ide/idea/scan/ScanBinaryExecutorTest.java

```diff
@@ -0,0 +1,90 @@
+package com.jfrog.ide.idea.scan;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.jfrog.ide.idea.scan.data.ScanConfig;
+import com.jfrog.ide.idea.scan.data.ScansConfig;
+import junit.framework.TestCase;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+
+/**
+ * @author tala
+ **/
+public class ScanBinaryExecutorTest extends TestCase {
+    private ScanBinaryExecutor scanner;
+    private final Path TEST_DEMO_OUTPUT = new File("src/test/resources/sarif/demo_output.sarif").toPath();
+
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        scanner = new ApplicabilityScannerExecutor();
+
+    }
+
+    public void testInputBuilder() {
+        var inputFileBuilder = new ScanConfig.Builder();
+        Path inputPath = null;
+        var testOutput = "file/location\\out.sarif";
+        var testLanguage = "Go";
+        var testRoots = List.of("a", "b", "c");
+
+        inputFileBuilder.scanType(scanner.SCAN_TYPE);
+        inputFileBuilder.output(testOutput);
+        inputFileBuilder.language(testLanguage);
+        inputFileBuilder.roots(testRoots);
+        try {
+            inputPath = scanner.createTempRunInputFile(new ScansConfig(List.of(inputFileBuilder.Build())));
+            var inputFile = readScansConfigYAML(inputPath);
+            assertNotNull(inputFile);
+            assertEquals(1, inputFile.getScans().size());
+            assertEquals(testOutput, inputFile.getScans().get(0).getOutput());
+            assertEquals(testLanguage, inputFile.getScans().get(0).getLanguage());
+            assertEquals(testRoots, inputFile.getScans().get(0).getRoots());
+
+        } catch (Exception e) {
+            assertNull(e);
```

Comment:
You can remove this catch clause. Any non-catched exception would fail the test.
Same in the other test.